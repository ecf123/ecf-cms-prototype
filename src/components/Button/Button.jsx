import React from "react";
import "./Button.scss";

const Button = ({
  style,
  textStyle,
  buttonText,
  children,
  imgStyle,
  buttonFunction,
}) => {
  return (
    <button className={style} onClick={buttonFunction}>
      <div className={imgStyle}>{children}</div>
      <p className={textStyle}> {buttonText}</p>
    </button>
  );
};

export default Button;
