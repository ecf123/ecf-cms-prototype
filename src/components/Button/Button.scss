@use "../../styles/variables" as *;

* {
  margin: 0;
  padding: 0;
}

.button {
  width: fit-content;
  text-align: center;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  &:active{
    opacity: 0.8;
  }
}

.black-svg {
  svg {
    path {
      stroke: $color-blue-navy-dark;
    }
  }
}

.white-svg {
  svg {
    path {
      stroke: $color-white;
    }
  }
}
.small-img {
  svg {
    display: flex;
    align-items: center;
    margin-right: 0.35rem;
    path {
      height: 0.47rem;
      width: auto;
    }
  }
}

.medium-img {
  svg {
    display: flex;
    align-items: center;
    path {
      height: 0.75rem;
      width: auto;
    }
  }
}

.large-img {
  margin-right: 0.44rem;
  display: flex;
  align-items: center;
  svg {
    path {
      height: 1.06rem;
      width: auto;
    }
  }
}

.no-img {
  width: 0;
}

.text {
  font-family: $font-secondary;
  font-weight: $font-weight-regular;
}

.round-border {
  border-radius: 2.5rem;
}

.square-border {
  border-radius: 0.26rem;
}

.rounded-square-border {
  border-radius: 0.5rem;
}

.green {
  background-color: $color-green-light;
  color: $background-color;
}

.grey {
  background-color: $color-grey-light;
  color: $background-color;
}

.light-grey {
  background-color: $color-white-smoke;
  color: $background-color;
}

.white {
  background-color: $color-white;
  color: $background-color;
}

.black {
  background-color: $color-blue-navy-dark;
  color: $color-white;
}

.border {
  border: 1px solid $color-grey-light;
}

.red {
  background-color: $color-red-normal;
  color: $color-white;
}

.small {
  height: 1.375rem;
  padding: 0.502rem 0.5rem;
}

.small-text {
  margin: 0;
  font-size: 0.6rem;
}

.medium {
  height: 2.45rem;
  padding: 0.6875rem 1.25rem;
}

.medium-text {
  margin: 0;
  font-size: $font-size-extra-small;
  margin-left: 0.45rem;
}

.large {
  height: 3.1rem;
  padding: 0.9375rem 2.5rem;
}

.large-text {
  margin: 0;
  font-size: $font-size-small;
  font-weight: $font-weight-bold;
}

.modal-text {
  font-size: $font-size-small;
  font-weight: $font-weight-regular;
}

.login-black {
  background-color: $color-green;
  padding: 0.844rem 10.063rem;
}

.login-green {
  background-color: #b6ddc2;
  padding: 1.0625rem 12.6875rem;
}

.modal {
  height: 2.75rem;
  padding: 0.625rem;
  width: 10.625rem;
}