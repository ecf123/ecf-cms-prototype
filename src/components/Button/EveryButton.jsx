import React from 'react'
import Button from './Button'
import './Button.scss'
import bin from "../../assets/bin-icon.svg";
import pencil from "../../assets/pen-icon.svg";
import edit from "../../assets/notepad-and-pencil-grey.svg";
import plus from "../../assets/circular-plus-icon.svg";


const EveryButton = () => {
  return (
    <>
    < Button style={"button red square-border small"} imgStyle={"small-img"} textStyle={"text small-text red"} buttonIcon={bin} buttonText={"Delete"}/>
    < Button style={"button grey square-border small"} imgStyle={"small-img"} textStyle={"text small-text grey"} buttonIcon={edit} buttonText={"Edit"}/>
    < Button style={"button black round-border medium"} imgStyle={"medium-img"} textStyle={"text medium-text black"} buttonIcon={pencil} buttonText={"Add Course"}/>
    < Button style={"button black round-border medium"} imgStyle={"large-img"} textStyle={"text large-text black"} buttonIcon={plus} buttonText={"Add Lesson"}/>
    < Button style={"button black round-border large"}textStyle={"text large-text black"} buttonText={"Submit"}/>
    < Button style={"button grey round-border large"}textStyle={"text large-text grey"} buttonText={"Cancel"}/>
    < Button style={"button red round-border medium"} imgStyle={"medium-img"} textStyle={"text medium-text red"} buttonIcon={bin} buttonText={"Delete"}/>
    < Button style={"button grey round-border small"}  textStyle={"text small-text grey"}  buttonText={"Upload Image"}/>
    < Button style={"button grey round-border medium"} imgStyle={"medium-img"} textStyle={"text medium-text grey"} buttonIcon={pencil} buttonText={"Edit"}/>
    < Button style={"button login-black square-border"}  textStyle={"text large-text black"} buttonText={"Login"}/>
    < Button style={"button login-green square-border"}  textStyle={"text large-text green"} buttonText={"Login"}/>
    < Button style={"button red rounded-square-border modal"} textStyle={"text modal-text red"} buttonText={"Delete"}/>
    < Button style={"button white rounded-square-border border modal"} textStyle={"text modal-text white"} buttonText={"Cancel"}/>
    </>
  )
}

export default EveryButton