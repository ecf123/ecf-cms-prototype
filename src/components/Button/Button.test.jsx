import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import Button from "./Button";

it("Should render the Button", () => {
  const { container } = customRender(<Button />);
  expect(container).toMatchSnapshot();
});

it("Should render the svg in the Button", () => {
  render(
    <Button>
      <p>This is an svg image</p>
    </Button>
  );
  const svg = screen.getByText("This is an svg image");
  expect(svg).toBeInTheDocument();
});

it("Should display the text in the button", () => {
  render(<Button buttonText="Cancel" />);
  const text = screen.getByText("Cancel");
  expect(text).toBeInTheDocument();
});
