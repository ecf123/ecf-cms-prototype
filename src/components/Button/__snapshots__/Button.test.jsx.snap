// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Button 1`] = `
<div>
  <button>
    <div />
    <p>
       
    </p>
  </button>
</div>
`;
